src/test/java/org/sonarlint/intellij/core/ProjectServerNotificationsSubscriberTest.kt

```diff
@@ -0,0 +1,165 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2020 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.core
+
+import com.intellij.notification.Notification
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.Captor
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.never
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.verifyZeroInteractions
+import org.mockito.junit.MockitoJUnitRunner
+import org.sonarlint.intellij.AbstractSonarLintLightTests
+import org.sonarlint.intellij.any
+import org.sonarlint.intellij.capture
+import org.sonarlint.intellij.config.global.ServerConnection
+import org.sonarlint.intellij.messages.GlobalConfigurationListener
+import org.sonarlint.intellij.messages.ProjectConfigurationListener
+import org.sonarlint.intellij.ui.BalloonNotifier
+import org.sonarsource.sonarlint.core.client.api.common.NotificationConfiguration
+
+@RunWith(MockitoJUnitRunner::class)
+class ProjectServerNotificationsSubscriberTest : AbstractSonarLintLightTests() {
+  private lateinit var serverNotificationsService: ServerNotificationsService
+  private val balloonNotifier = mock(BalloonNotifier::class.java)
+  private lateinit var projectServerNotificationsSubscriber: ProjectServerNotificationsSubscriber
+
+  @Captor
+  private lateinit var notificationConfigurationCaptor: ArgumentCaptor<NotificationConfiguration>
+
+  @Captor
+  private lateinit var balloonNotificationCaptor: ArgumentCaptor<Notification>
+
+  @Before
+  fun setup() {
+    replaceProjectService(BalloonNotifier::class.java, balloonNotifier)
```

Comment:
Instead of mocking notification, another option is to let the real component, and add a listener: https://github.com/jansorg/intellij-notifications/blob/master/src/test/java/dev/ja/samples/ConfigurableNotificationTest.java

Reply:
Good to know! I added this BalloonNotifier some time ago to be able to mock and check the call to notify. I will try to replace with this mechanism